### ISSUE TITLE
RFC 0511 - challenge/domain should be not required (or optional)

### ISSUE THREAD

**Issue by awoie** [hyperledger/aries-rfcs#538]
The current spec says that offer-credential `challenge` and `domain` are REQUIRED.

There might be two issues:
- It contradicts with the request-credential message where the spec says that `challenge` and `domain` might be obtained OOB. If OOB transmission is allowed, then we should probably not require these two fields in the request-credential message.
- The DIF Credential Manifest specification can include a `presentation_definition` which is defined [here](https://identity.foundation/presentation-exchange/#presentation-definition). Further, the Presentation Definition MAY include a Presentation Request which allows to include `challenge` / `domain` parameter. So, instead of passing `challenge` / `domain` in the offer-credential as individual parameters, we could use the Presentation Request to pass `challenge` / `domain` to the Holder for the Presentation Submission.

**Comment by llorllale** [comment]
@awoie 

> It contradicts with the request-credential message where the spec says that challenge and domain might be obtained OOB. If OOB transmission is allowed, then we should probably not require these two fields in the request-credential message.

I agree. Another reason is the fact that a CM may _not_ have a `presentation_definition`. 

> The DIF Credential Manifest specification can include a presentation_definition which is defined here. Further, the Presentation Definition MAY include (or MAY be the same as--still unclear to me) a Presentation Request which allows to include challenge / domain parameters. So, instead of passing challenge / domain in the offer-credential as individual parameters, we could use the Presentation Request to pass challenge / domain to the Holder for the Presentation Submission.

As I answered in https://github.com/decentralized-identity/presentation-exchange/issues/112#issuecomment-698348423, a presentation definition is not the same as a presentation request. This means _if you are using issue-credential v2 with RFC0511 alone_ then both `domain` and `challenge` need to be send in `offer-credential`.

I haven't fully explored the alternative of running present-proof as a co-protocol in order to deliver the proof to the issuer but it seems to have unnecessary overhead for this use case IMO. The Holder would need to send a `propose-presentation` msg in the role of Prover, the Issuer would need to respond with a `request-presentation` in the role of Verifier, and the Holder would respond with `presentation` in the role of Prover.
